### DISCUSSION TITLE
High Priority: Integration Tally with Darwinia/Crab

### DISCUSSION THREAD

**Discussion by hackfisher** [ringdao-community-guild/discussion#87] [Integration]
https://docs.tally.xyz/user-guides/tally-contract-compatibility/network-support

@hujw77 needs to involved in this work to provide guidance

**Comment by hujw77** [comment]
To support a chain, Tally has several must-have dependencies:
EVM-compatible Network
- [x] A full node for user web3 calls  cc @fewensa @boundless-forest
- [x] An archive node with a high rate limit for indexing cc @fewensa @boundless-forest
- [x] A block explorer with an Etherscan-compatible API cc @freehere107

First, we should meet their must-have dependencies.
Can we confirm if Darwinia and Crab network meet these conditions?
@fewensa @boundless-forest @freehere107

**Reply by freehere107** [reply]
@hujw77 Can you help confirm which APIs require Etherscan-compatible? It would be best if you could list them.

**Reply by hujw77** [reply]
Please refer [EIP-3091](https://eips.ethereum.org/EIPS/eip-3091). cc @freehere107

**Reply by freehere107** [reply]
@hujw77 https://eips.ethereum.org/EIPS/eip-3091 subscan is supported

**Reply by hujw77** [reply]
https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-46.json
https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-44.json
Please update this info. change `none` to `EIP3091`.  @boundless-forest
```
  "explorers": [
    {
      "name": "subscan",
      "url": "https://darwinia.subscan.io",
      "standard": "none"
    }
  ]
  ```

**Reply by boundless-forest** [reply]
Got it. I'll update this today. Then feedback here.

**Reply by fewensa** [reply]
1. Our Darwinia and Crab RPC are Full node. and I notified to all community rpc [upgrade to sql backend](https://github.com/dcdao/community-rpc/issues/28). 
2. There is currently no request limit. If you have any questions, please contact me.

**Reply by boundless-forest** [reply]
The explorers information has been updated. See https://github.com/ethereum-lists/chains/pull/4403

**Comment by hujw77** [comment]
Hey @hackfisher, we already have the capability to integrate Tally. The next step is to contact the project team for integration Darwinia and Crab, although ensuring the RPC rate limit of the Crab network may not be guaranteed. 
I have already submitted a request to integrate Tally on Discord, but their response has been rather slow. https://discord.com/channels/783459824029007882/1209067464558776361

**Reply by hujw77** [reply]
![Screenshot 2024-02-29 at 10 02 21](https://github.com/dcdao/discussion/assets/6929585/0d4a5a18-2576-46df-b680-06750486ae71)

**Reply by hujw77** [reply]
![Screenshot 2024-03-05 at 09 32 06](https://github.com/dcdao/discussion/assets/6929585/544f37e3-6191-45f1-a670-6a43c358edc8)

**Reply by hujw77** [reply]
![Screenshot 2024-03-10 at 21 22 44](https://github.com/dcdao/discussion/assets/6929585/723d9cd9-e5aa-425f-8cca-aa739da3d8f7)

**Reply by hujw77** [reply]
![Screenshot 2024-04-02 at 22 30 38](https://github.com/dcdao/discussion/assets/6929585/97ccad37-358a-4df5-8dab-a865be73eeb0)


![Screenshot 2024-04-02 at 22 26 57](https://github.com/dcdao/discussion/assets/6929585/ba7fa09d-ac22-43bf-aad5-417d87390b77)
